apps/safe-claiming-app/src/Widget.tsx

```diff
@@ -0,0 +1,191 @@
+import { useDelegatesFile } from "src/hooks/useDelegatesFile"
+import { useFetchVestingStatus } from "src/hooks/useFetchVestingStatus"
+import { useTokenBalance } from "src/hooks/useTokenBalance"
+import { sameAddress } from "src/utils/addresses"
+import { formatAmount } from "src/utils/format"
+
+const SpaceContent = styled("div")`
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+`
+
+const Title = (props: TypographyProps) => (
+  <Typography
+    color="primary.main"
+    style={{ fontWeight: "bold", textAlign: "center" }}
+  >
+    {props.children}
+  </Typography>
+)
+
+const Subtitle = (props: TypographyProps) => (
+  <Typography
+    variant="subtitle2"
+    color="primary.light"
+    style={{ marginBottom: "16px", textAlign: "center" }}
+  >
+    {props.children}
+  </Typography>
+)
+
+const StyledButton = (props: ButtonProps) => (
+  <Button
+    size="large"
+    onClick={props.onClick}
+    variant="contained"
+    disableElevation
+  >
+    {props.children}
+  </Button>
+)
+
+const Widget = () => {
+  const [delegates] = useDelegatesFile()
+  const delegateAddressFromContract = useDelegate()
+  const balance = useTokenBalance()
+  const { safe } = useSafeAppsSDK()
+
+  const [vestings] = useAirdropFile()
+  const [userVesting, ecosystemVesting, investorVesting] = vestings
+
+  const totalAllocation = BigNumber.from(userVesting?.amount || 0)
+    .add(ecosystemVesting?.amount || 0)
+    .add(investorVesting?.amount || 0)
+
+  const currentDelegate = useMemo(() => {
+    if (delegateAddressFromContract) {
+      const registeredDelegateFromData = delegates.find((entry) =>
+        sameAddress(entry.address, delegateAddressFromContract)
+      )
+      return (
+        registeredDelegateFromData || { address: delegateAddressFromContract }
+      )
+    }
+  }, [delegateAddressFromContract, delegates])
+
+  const [userVestingStatus] = useFetchVestingStatus(
+    userVesting?.vestingId,
+    userVesting?.contract
+  )
+
+  const [ecosystemVestingStatus] = useFetchVestingStatus(
+    ecosystemVesting?.vestingId,
+    ecosystemVesting?.contract
+  )
+
+  const [investorVestingStatus] = useFetchVestingStatus(
+    investorVesting?.vestingId,
+    investorVesting?.contract
+  )
+
+  const totalClaimed = BigNumber.from(userVestingStatus?.amountClaimed || 0)
+    .add(ecosystemVestingStatus?.amountClaimed || 0)
+    .add(investorVestingStatus?.amountClaimed || 0)
+
+  const votingPower = useMemo(() => {
+    if (totalAllocation.gt(0) && totalClaimed.gt(0) && balance.gt(0)) {
+      return totalAllocation.sub(totalClaimed).add(balance)
+    }
+    return BigNumber.from(0)
+  }, [balance, totalAllocation, totalClaimed])
+
+  const handleClickClaim = () => {
+    const url = `${window.location.ancestorOrigins[0]}/apps?safe=${
+      safe.chainId === 1 ? "eth" : "gor"
+    }:${safe.safeAddress}&appUrl=${CLAIMING_DATA_URL}`
+
+    // @ts-ignore
+    window.top.location.href = url
```

Comment:
I think it is quite dangerous, that this is currently possible from our Safe Apps as this is enables Safe apps to redirect our parent window. This could potentially lead to redirects to phishing / scam version of our interface from safe apps. 

In my opinion we should add some restrictions to our Safe apps by using the sandbox attribute of the iframe.
Some more info: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attr-sandbox
e.g.
```html
<iframe sandbox="allow-scripts" />
```